core/backend/src/CloudSqlite.ts

```diff
@@ -623,13 +623,16 @@ export namespace CloudSqlite {
     await acquireWriteLock(args);
```

Comment:
I think it's easier to just do:

```ts
    if (args.container.hasWriteLock)
      return operation();
```

Reply:
if (args.container.hasWriteLock && heldBySameUser) ?

Reply:
Also, if I'm not mistaken the current state of the code does not allow you to refresh the write lock if you're the same user in the case of a long-running operation? CloudSqlite.acquireWriteLock returns if its the same user, but the native version of acquireWriteLock has

 "Calling `acquireWriteLock` with the lock already held resets the lock duration from the current time,
     * so long running processes should call this method periodically to ensure their lock doesn't expire (they should also make sure their accessToken is refreshed
     * before it expires.)"

Reply:
there's only one write lock, so if it's already held we should do what the comment says. But the whole "same user" thing is only relevant for RPC calls, which can't be recursive or long-running.